api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -293,4 +296,19 @@ public Provider getUnknownProvider() {
 		return getProviderByUuid(Context.getAdministrationService().getGlobalProperty(
 		    OpenmrsConstants.GP_UNKNOWN_PROVIDER_UUID));
 	}
+
+	@Override
+	public Provider createProviderFromUser(User user){
+		Provider p = new Provider();
+		if (user == null || user.getPerson() == null){
+			throw new APIException("User cannot be null");
+		} else if (Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()){
+			p.setPerson(user.getPerson());
+			return Context.getProviderService().saveProvider(p);
+		} else {
```

Comment:
Maybe you can avoid using `else` statement here, since it is unnecessary (refer to [this](https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-CodeStyle)). Rather you can simply write:
```
    if (user == null || user.getPerson() == null){
		throw new APIException("User cannot be null");
	} else if (Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()){
		p.setPerson(user.getPerson());
		return Context.getProviderService().saveProvider(p);
	} 
	log.warn("Provider already exists for user");
	return
	Context.getProviderService().getProvidersByPerson(user.getPerson()).iterator().next();
```

Reply:
Also, this is the part (`else if` and `else`) where you're missing out on code coverage, and hence the check failure.